### PR TITLE
Update kubernetes submodule, generated files, and vendored files to v1.24.3 with required fixes to tests and code

### PR DESCRIPTION
**Pull request by markstur**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Need to update kubernetes submodule, generated files, and vendored files.  This PR updates to v1.24.3 and updates calls and tests as required by the new version.

- Update simulator/submodule/kubernetes to v1.24.3
- Update go.mod to v1.24.3 and run go mod tidy
- go mod vendor to update simulator/vendor to v1.24.3
- go generate files with fixes for version update and lint
- Updates mock expects where new version changed calls (PVC behavior)
- Adds DynamicSharedInformerFactory where required

#### Which issue(s) this PR fixes:

Part of #46 

#### Special notes for your reviewer:

This was part of #46, but after consideration these updates should be an independent PR (also probably another PR for the next bump to v1.25) before touching the config version and especially before addressing multipoint.

/label tide/merge-method-squash

### CODE REVIEW

**Comment by markstur** [comment]
@sanposhiho this update would be a prereq to fixing #46

**Review by sanposhiho — COMMENTED** [review]
Thanks! Left one review but looks good overall.

File: simulator/scheduler/scheduler.go [hunk 1]
```diff
@@ -63,29 +65,32 @@ func (s *Service) StartScheduler(versionedcfg *v1beta2config.KubeSchedulerConfig
...
-
 	evtBroadcaster.StartRecordingToSink(ctx.Done())
 
 	s.currentSchedulerCfg = versionedcfg.DeepCopy()
 
 	cfg, err := convertConfigurationForSimulator(versionedcfg)
 	if err != nil {
-		cancel()
-		return xerrors.Errorf("convert scheduler config to apply: %w", err)
+		return cancelAndErrorf(cancel, "convert scheduler config to apply: %w", err)
```
[the remark this answers is not in the document]
Comment (markstur) [reply simulator/scheduler/scheduler.go]: Thanks! That is a much better way to do cancel() on exit-with-error.  I just pushed up that change.

Actually, my reason for consolidating cancelAndErrorf was just to make lint funlen (function length) happy, and that's where I found I could refactor/reuse to save a few lines. Turns out defer func... with proper formatting costs a few lines so I got rid of more white space.  A bit unfortunate, but I think there is some benefit in how it "fits a screen" better.

So, I like your suggestion and think we should keep it for the cancel() thing.  If the formatting change bothers anyone, I could put some of the old spacing back in and just turn off lint(funlen).

**Review by sanposhiho — APPROVED** [review]
/approve
/lgtm

Thanks!

**Pull request merged**